proxy/proxy.go

```diff
@@ -0,0 +1,30 @@
+package proxy
+
+import (
+	"net"
+	"net/http"
+	"time"
+)
+
+// MakeHTTPClient makes a HTTP client with good defaults for timeouts.
+func MakeHTTPClient(timeout *time.Duration) http.Client {
+	if timeout != nil {
```

Comment:
Is this meant to be `timeout`'s only purpose?

Reply:
How do you mean?